hera_sim/simulate.py

```diff
@@ -434,4 +451,69 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
                 vis=self.data.data_array[blt_ind, 0, :, pol_ind],
                 xtalk=xtalk
             )
+    
+    
+    def run_sim(self, sim_file=None, **sim_params):
+        """
+        Accept a dictionary or YAML file of simulation parameters and add in
+        all of the desired simulation components to the Simulator object.
+
+        Args:
+            sim_file (str, optional): string providing a hook to a YAML file
+            
+            sim_params (dict, optional): dictionary of simulation parameters.
+                it must take the form {model:params}, where model is
+                a string specifying the simulation component to be added
+                and params is a dictionary providing all the keyword
+                arguments and their desired values.
+
+            One (and *only* one) of the above arguments must be provided. If 
+            *both* sim_file and sim_params are provided, then this function
+            will raise an AssertionError.
+        """
+
+        # keep track of which components don't use models
+        uses_no_model = []
+        for key, val in self.SIMULATION_COMPONENTS.items():
+            func = getattr(self, val)
+            if 'model' not in inspect.signature(func).parameters:
```

Comment:
This bit of code is breaking in the Python 2.7 build. Apparently the behavior of `inspect.signature` is quite different between the Python 2 and Python 3 implementations. When this line is run in the Python 2.7 build, an error is thrown, stating something along the lines of "modules don't have signatures".

Is it important to fix this to keep Python 2 compatibility? If so, how can we go about fixing it?

Reply:
Ah yeah, should've thought of that. I'm inclined to say we drop Py2 support -- in <6 months we'll be forced to anyway. So as not to cause a fuss, you can probably just do something like 

```
if sys.version_info < (3, 0):
    <python2 way of doing inspect.signature (look it up)>
else:
   < what you have now>
```

Reply:
sounds good

Reply:
After a bit of digging around, I've come to the conclusion that there isn't a simple workaround for Py2. The main issue seems to be that `functools.wraps` preserves signatures for python versions 3.4 and later, but it does not for earlier versions of python. So the `_model` decorator overwrites the function signatures when `hera_sim` is used on Py2. It looks like there *is* a solution, but it involves rewriting the `_model` decorator using the `decorator` module.

Is it worth doing all this extra work to support a soon-to-be-deprecated version of python?

Reply:
Most definitely not. I forgot that it was decorated. You could just raise an error if python 2 is being used for this particular method.

Reply:
This looks good